lib/srv/db/server.go

```diff
@@ -200,12 +191,44 @@ func (s *Server) initDatabaseServer(ctx context.Context, server types.DatabaseSe
 	if err := s.initHeartbeat(ctx, server); err != nil {
 		return trace.Wrap(err)
 	}
-	if err := s.initRDSRootCert(ctx, server); err != nil {
+	if err := s.initCACert(ctx, server); err != nil {
+		return trace.Wrap(err)
+	}
+	if err := s.initCloudClients(ctx, server); err != nil {
 		return trace.Wrap(err)
 	}
 	return nil
 }
 
+func (s *Server) initCloudClients(ctx context.Context, server types.DatabaseServer) error {
+	var err error
+	switch server.GetType() {
+	case types.DatabaseTypeRDS, types.DatabaseTypeRedshift:
+		region := server.GetAWS().Region
+		if _, ok := s.awsSessions[region]; ok {
+			return nil // Already initialized.
+		}
+		s.awsSessions[region], err = awssession.NewSessionWithOptions(awssession.Options{
+			SharedConfigState: awssession.SharedConfigEnable,
+			Config: aws.Config{
+				Region: aws.String(region),
+			},
+		})
+		if err != nil {
+			return trace.Wrap(err)
+		}
```

Comment:
Map entry will be created on error case, meaning that the above condition does not hold.

Reply:
Nice catch!